deliver/spec/commands_generator_spec.rb

```diff
@@ -0,0 +1,242 @@
+      expect(Deliver::GenerateSummary).to receive(:new).and_return(fake_generate_summary)
+      expect(fake_generate_summary).to receive(:run) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      Deliver::CommandsGenerator.start
+    end
+
+    it "can use the app_identifier flag from tool options" do
+      stub_commander_runner_args(['generate_summary', '--description', 'My description', '--app_identifier', 'abcd', '-f', 'true'])
+
+      expected_options = FastlaneCore::Configuration.create(Deliver::Options.available_options, {
+        description: 'My description',
+        app_identifier: 'abcd',
+        force: true
+      })
+
+      expect(Deliver::Runner).to receive(:new) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      fake_generate_summary = "generate_summary"
+      expect(Deliver::GenerateSummary).to receive(:new).and_return(fake_generate_summary)
+      expect(fake_generate_summary).to receive(:run) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      Deliver::CommandsGenerator.start
+    end
+  end
+
+  describe ":download_screenshots option handling" do
+    it "can use the username short flag from tool options" do
+      stub_commander_runner_args(['download_screenshots', '--description', 'My description', '-u', 'me@it.com', '-w', 'screenshots/path'])
+
+      expected_options = FastlaneCore::Configuration.create(Deliver::Options.available_options, {
+        description: 'My description',
+        username: 'me@it.com',
+        screenshots_path: 'screenshots/path'
+      })
+
+      expect(Deliver::Runner).to receive(:new) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      expect(Deliver::DownloadScreenshots).to receive(:run) do |actual_options, screenshots_path|
+        expect(expected_options._values).to eq(actual_options._values)
+        expect(screenshots_path).to eq('screenshots/path')
+      end
+
+      Deliver::CommandsGenerator.start
+    end
+
+    it "can use the app_identifier flag from tool options" do
+      stub_commander_runner_args(['download_screenshots', '--description', 'My description', '--app_identifier', 'abcd', '-w', 'screenshots/path'])
+
+      expected_options = FastlaneCore::Configuration.create(Deliver::Options.available_options, {
+        description: 'My description',
+        app_identifier: 'abcd',
+        screenshots_path: 'screenshots/path'
+      })
+
+      expect(Deliver::Runner).to receive(:new) do |actual_options|
+        expect(expected_options._values).to eq(actual_options._values)
+      end
+
+      expect(Deliver::DownloadScreenshots).to receive(:run) do |actual_options, screenshots_path|
+        expect(expected_options._values).to eq(actual_options._values)
+        expect(screenshots_path).to eq('screenshots/path')
+      end
```

Comment:
These few lines also look to be the same as 180 -183. Thoughts on pulling this out?